File: FLL Control/src/Match.java
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;


public class Match {
	private List<Team> teams;
	private Map<Integer, List<Team>> tableAssignments;
	
	public Match() {
		teams = new LinkedList<Team>();
		tableAssignments = new HashMap<>();
	}
	
	public void addTeam(Team team, Integer table) {
		teams.add(team);
		if(!tableAssignments.containsKey(table))
			tableAssignments.put(table, new LinkedList<Team>());
		tableAssignments.get(table).add(team);
	}
	
	public List<Team> getTeams() {
		return teams;
	}
	
	public List<Team> getTeamsOnTable(int table) {
		return tableAssignments.get(table);
	}
}

File: FLLScoreKeeper/FLLShared/match.cpp
#include "match.h"

using namespace std;

void Match::addTeam(shared_ptr<Team> team, int table) {
    teams.push_back(team);
    table_assignments[table].push_back(team);
}

const vector<shared_ptr<Team>>& Match::getTeams() const {
    return teams;
}

const vector<shared_ptr<Team>>& Match::getTeamsOnTable(int table) const {
    return table_assignments.at(table);
}
